### PR TITLE
mgr/dashboard: Remove Histogram Component

### PR DESCRIPTION
**Pull request by arbtfnf**
Removed the osd-details histogram component and the related code

Screenshot from 2020-04-15 14-17-24

Fixes: <https://tracker.ceph.com/issues/26954> 
Signed-off-by: Anurag Bandhu <abandhu@redhat.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by arbtfnf** [comment]
jenkins test dashboard

**Comment by arbtfnf** [comment]
> I would suggest the title as `mgr/dashboard: "Whatever isssue the PR fixes" ` as the title of the PR (same goes for the commit message). Also it would be good to include some screenshots of the output.

Thank you, for the feedback.

**Comment by s0nea** [comment]
Hey Anurag,

If I'm wrong correct me please, but I think you need to sign your commit with your full name. Could you please check if this is the case for your commit? Thanks :)

**Comment by arbtfnf** [comment]
> Hey Anurag,
> 
> If I'm wrong correct me please, but I think you need to sign your commit with your full name. Could you please check if this is the case for your commit? Thanks :)
Sure, will do that.

**Comment by arbtfnf** [comment]
> Hey Anurag,
> 
> If I'm wrong correct me please, but I think you need to sign your commit with your full name. Could you please check if this is the case for your commit? Thanks :)

Sure, will do that

**Comment by nizamial09** [comment]
> > Hey Anurag,
> > If I'm wrong correct me please, but I think you need to sign your commit with your full name. Could you please check if this is the case for your commit? Thanks :)
> 
> Sure, will do that

While you are doing that please fix the commit title and message.

**Comment by nizamial09** [comment]
LGTM, Works fine for me

**Comment by nizamial09** [comment]
One question. Is the color for the Histogram is finalised or is it open for discussion? @arbtfnf

**Comment by LenzGr** [comment]
As per our conversation during the standup today, we concluded that the Histogram in it's current form does not really provide any useful information, regardless of the color scheme used. It was decided that we should rather remove the current implementation in its entirety and try to come up with a better solution that visualizes this type of information.

**Comment by epuertat** [comment]
@arbtfnf shouldn't we reopen this and simply remove the histogram component?

**Comment by arbtfnf** [comment]
jenkins test make check

**Comment by arbtfnf** [comment]
jenkins test dashboard

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by callithea** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
Any update on this? You need to remove all references to the component and some now unnecessary code.

**Comment by epuertat** [comment]
@aaSharma14 just checked the commits in this PR and I see a couple of merge commits over there. That's quite unusual in a PR. Could it be perhaps because you have 'git-pulled' your branch with a merge (default) strategy vs. a rebase (`git pull --rebase` to ensure your commits are always at the tip of the branch)?

**Comment by aaSharma14** [comment]
> @aaSharma14 just checked the commits in this PR and I see a couple of merge commits over there. That's quite unusual in a PR. Could it be perhaps because you have 'git-pulled' your branch with a merge (default) strategy vs. a rebase (`git pull --rebase` to ensure your commits are always at the tip of the branch)?

Yes, Its because I didn't pulled it with rebase..will keep in mind going forward

**Comment by aaSharma14** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
lgtm, just missing rebasing and squashing all commits.

**Comment by aaSharma14** [comment]
jenkins test dashboard

**Comment by aaSharma14** [comment]
jenkins test dashboard

**Comment by aaSharma14** [comment]
> Rebase and squash all commits.

@tspmelo Done.

File: src/pybind/mgr/dashboard/controllers/osd.py [hunk 1]
```diff
@@ -99,22 +99,11 @@ def get(self, svc_id):
...
-        except SendCommandError as e:  # pragma: no cover - the handling is too obvious
-            if 'osd down' in str(e):  # pragma: no cover - no complexity there
-                histogram = str(e)
-            else:  # pragma: no cover - no complexity there
-                raise
-
         return {
             'osd_map': self.get_osd_map(svc_id),
             'osd_metadata': mgr.get_metadata('osd', svc_id),
-            'histogram': histogram,
```
Comment (votdev) [inline src/pybind/mgr/dashboard/controllers/osd.py:117]: I don't think this is a good idea to remove this now that the Dashboard REST API is the official Ceph REST API. Maybe this should be discussed in the next Dashboard standup meeting.

Comment (epuertat) [reply src/pybind/mgr/dashboard/controllers/osd.py:117]: Thanks for pointing this out @votdev. As discussed in today's stand-up, we may get rid of this (also saving one `send_command()` call per OSD every 5 secs) and move it to a separate end-point (`api/osd/{id}/histogram`): https://tracker.ceph.com/issues/46898

**Comment by tspmelo** [comment]
Please change the commit title from "`mgr/dashboard: Histogram on OSD view lacks description`" to something like "`mgr/dashboard: Remove Histogram Component`"

**Comment by aaSharma14** [comment]
jenkins test api

**Comment by aaSharma14** [comment]
jenkins test api

**Comment by aaSharma14** [comment]
jenkins test api

**Comment by aaSharma14** [comment]
> Please change the commit title from "`mgr/dashboard: Histogram on OSD view lacks description`" to something like "`mgr/dashboard: Remove Histogram Component`"

Done.! @tspmelo

**Comment by aaSharma14** [comment]
jenkins test api

**Comment by aaSharma14** [comment]
jenkins test api

**Comment by aaSharma14** [comment]
jenkins test api

**Comment by callithea** [comment]
- [pkg build](https://shaman.ceph.com/builds/ceph/wip-laura-testing-34644-36347/)
- [QA run](https://pulpito.ceph.com/laura-2020-08-20_12:57:25-rados:dashboard-wip-laura-testing-34644-36347-distro-basic-smithi/) [WIP]

**Review by callithea — APPROVED** [review]

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-details/osd-details.component.html [hunk 2]
```diff
@@ -50,30 +50,7 @@
...
-          <div class="col-md-6">
-            <h4 i18n>Reads</h4>
-            <cd-osd-performance-histogram [histogram]="osd?.details?.histogram?.osd?.op_r_latency_out_bytes_histogram">
-            </cd-osd-performance-histogram>
-          </div>
-        </div>
-      </ng-template>
-    </li>
-    <li ngbNavItem="performance-details"
+    <li ngbNavItem
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-details/osd-details.component.html:53]:
```suggestion
    <li ngbNavItem="performance-details"
```

**Review by tspmelo — CHANGES_REQUESTED** [review]

**Comment by aaSharma14** [comment]
jenkins test dashboard

**Review by tspmelo — APPROVED** [review]

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by aaSharma14** [comment]
jenkins test dashboard

**Comment by aaSharma14** [comment]
jenkins test dashboard

**Comment by aaSharma14** [comment]
jenkins test dashboard

**Pull request merged**